ConvivaTestApp/src/androidTest/java/com/bitmovin/analytics/conviva/testapp/AdsTrackingTests.kt

```diff
@@ -22,6 +22,7 @@ class AdsTrackingTests : TestBase() {
 
         // load source and verify
         loadSource(activityScenario, DEFAULT_DASH_VOD_SOURCE)
+        Thread.sleep(2000)
```

Comment:
Why do you need this sleep now? 🤔

Reply:
sleep() was earlier inside the activityScenario block in loadSource() method. When testing repeatedly with V3 upgrade, it turned out that the test results are not consistent. Moving the sleep outside of activityScenario blocks in all cases improved test result consistency, hence this change.

Reply:
Without a deeper look that indicates something "wrong". Apart from that a sleep of 2 secs seems a bit high.

Reply:
You are right. Tests pass with even 1 second sleep but occasionally these failed with with shorter sleep specially on Android simulator. So I added 2 seconds for stability of test results. Do you foresee any concerns around keeping it 2 seconds?

Reply:
Hm, ok I can see why you need to sleep here (waiting for the `onActivity` callback).

However I'd still prefer leaving this sleep in the method where it is actually "needed" i.e. the `loadSource` call.

I also hear you on that this improves stability but usually something like this indicates async work, and maybe there is a better solution to it than putting 2 secs sleeps all over the place. 

> Do you foresee any concerns around keeping it 2 seconds?

basically no (apart from the mentioned already) but it will increase your test run time significantly. _Yes it has been like this before, but if you can fix/find the actual root cause it may increase your test run times a lot_

But as I do not know enough on your test setup and all the implications, I leave it up to you as the PR creator in the end on how you proceed with this.

Reply:
moved Sleep call to load method